src/ValidatingHydrator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator\Validator;
+
+use Yiisoft\Hydrator\HydratorInterface;
+use Yiisoft\Hydrator\Validator\Attribute\ValidateResolver;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\ValidatorInterface;
+
+final class ValidatingHydrator implements HydratorInterface
+{
+    public function __construct(
+        private HydratorInterface $hydrator,
+        private ValidatorInterface $validator,
+        private ValidateResolver $validateResolver,
+    ) {
+    }
+
+    public function hydrate(
+        object $object,
+        array $data = [],
+        array $map = [],
+        bool $strict = false
+    ): void {
+        $result = $this->beforeAction();
+        $this->hydrator->hydrate($object, $data, $map, $strict);
+        $this->afterAction($object, $result);
+    }
+
+    public function create(
```

Comment:
Agree. In separate PR add docs.

Reply:
It's method `create()` call once, because for hydrate/create object used another hydrator (see parameter in constructor), but no `ValidatingHydrator`.